packages/mini-browser/src/browser/mini-browser.ts

```diff
@@ -0,0 +1,593 @@
+        return button;
+    }
+
+    protected createNext(parent: HTMLElement): HTMLElement {
+        const button = this.onClick(this.createButton(parent, 'Show The Next Page', MiniBrowser.Styles.NEXT), this.navigateForwardEmitter);
+        return button;
+    }
+
+    protected createRefresh(parent: HTMLElement): HTMLElement {
+        const button = this.onClick(this.createButton(parent, 'Reload This Page', MiniBrowser.Styles.REFRESH), this.refreshEmitter);
+        return button;
+    }
+
+    protected createOpen(parent: HTMLElement): HTMLElement {
+        const button = this.onClick(this.createButton(parent, 'Open In A New Window', MiniBrowser.Styles.OPEN), this.openEmitter);
+        return button;
+    }
+
+    protected createButton(parent: HTMLElement, title: string, ...className: string[]): HTMLElement {
+        const button = document.createElement('div');
+        button.title = title;
+        button.classList.add(...className, MiniBrowser.Styles.BUTTON);
+        parent.appendChild(button);
+        return button;
+    }
+
+    // tslint:disable-next-line:no-any
+    protected onClick(element: HTMLElement, emitter: Emitter<any>): HTMLElement {
+        this.toDispose.push(addEventListener(element, 'click', () => {
+            if (!element.classList.contains(MiniBrowser.Styles.DISABLED)) {
+                emitter.fire(undefined);
+            }
+        }));
+        return element;
+    }
+
+    protected mapLocation(location: string): Promise<string> {
+        return this.locationMapper.map(location);
+    }
+
+    protected setInput(value: string) {
+        if (this.input.value !== value) {
+            this.input.value = value;
+        }
+    }
+
+    protected frameSrc() {
+        let src = this.frame.src;
+        try {
+            const { contentWindow } = this.frame;
+            if (contentWindow) {
+                src = contentWindow.location.href;
+            }
+        } catch {
+            // CORS issue. Ignored.
+        }
+        if (src === 'about:blank') {
+            src = '';
+        }
+        return src;
+    }
+
+    protected contentDocument(): Document | null {
+        try {
+            let { contentDocument } = this.frame;
+            if (contentDocument === null) {
+                const { contentWindow } = this.frame;
+                if (contentWindow) {
+                    contentDocument = contentWindow.document;
+                }
+            }
+            return contentDocument;
+        } catch {
+            // tslint:disable-next-line:no-null-keyword
+            return null;
+        }
+    }
+
+    protected async go(location: string | undefined, register: boolean = false): Promise<void> {
+        if (location) {
+            try {
+                const url = await this.mapLocation(location);
+                this.setInput(url);
+                this.showLoadIndicator();
```

Comment:
Can we delay this a bit? When editing an html file, the reload is very flickery, because of the intermediate reload screen.

Reply:
Sure. How many `ms` should be the delay?

Reply:
I will try it and adjust to having the best user experience.